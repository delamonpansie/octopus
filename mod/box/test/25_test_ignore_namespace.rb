#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[1].enabled = 1
object_space[1].index[0].type = "HASH"
object_space[1].index[0].unique = 1
object_space[1].index[0].key_field[0].fieldno = 0
object_space[1].index[0].key_field[0].type = "STR"
EOD
  end
end

Env.connect_eval do |server|

  insert ["foo"]
  insert ["foo"], :object_space => 1

  server.snapshot

  insert ["bar"]
  insert ["bar"], :object_space => 1

  select "foo", "bar"
  select "foo", "bar", :object_space => 1

  server.stop
  File.open(Env::ConfigFile, "a") { |io| io.puts("object_space[1].ignored = 1") }
  server.start

  wait_for "reconnect" do reconnect end

  insert ["baz"]
  log_try { insert ["baz"], :object_space => 1 }
  puts

  select "foo", "bar", "bar"
  select "foo", "bar", "baz", :object_space => 1
end
