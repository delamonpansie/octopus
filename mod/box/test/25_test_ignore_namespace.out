# box.insert(["foo"])
1

# box.insert(["foo"], {:object_space=>1})
1

# box.insert(["bar"])
1

# box.insert(["bar"], {:object_space=>1})
1

# box.select("foo", "bar")
[["foo"], ["bar"]]

# box.select("foo", "bar", {:object_space=>1})
[["foo"], ["bar"]]

# box.insert(["baz"])
1

# box.insert(["baz"], {:object_space=>1})
Failed with: {code: 0x202, message: 'ignored object space'}

# box.select("foo", "bar", "bar")
[["foo"], ["bar"], ["bar"]]

# box.select("foo", "bar", "baz", {:object_space=>1})
[]

